src/olympia/translations/fields.py

```diff
@@ -227,6 +227,8 @@ def __init__(self, *args, **kwargs):
         kwargs.pop('blank', None)
 
         super().__init__(*args, **kwargs)
+        # Some fields (e.g. EmailField) set a default we don't want to use
+        self.max_length = kwargs.get('max_length')
```

Comment:
Maybe we could raise an exception if it's not passed in `kwargs` ?

Reply:
it's not passed in kwargs in the general case (for our codebase) - because `form_field.max_length is None` the `max_length` from the model field is used instead. (disclaimer - I didn't manage to track down where the fallback logic about `None` values was, but we only define `100` in the model field so I concluded that's what is happening)

Reply:
The first fix I tried was just passing `max_length` in the form field kwargs -  `test_support_email_too_long` passes, and the other assert in `test_render_maxlength` passes, but `doc('input#id_support_email_0').attr('maxlength') is None` for some reason.

That's probably a bug with how we render the `TransMulti` widgets so again we're back to how much time we want to spend fixing our buggy translation handling code in devhub...

Reply:
Oh, the fallback logic is in _our_ codebase, not django's in `set_max_length_on_fields_where_necessary` - so I guess it's possible to tinker with that.

Reply:
< time we want to spend on fixing it > argument remains though.

Reply:
added a comment